src/commands/User.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonInteraction,
+  ButtonStyle,
+  EmbedBuilder,
+  SlashCommandBuilder,
+  type ChatInputCommandInteraction,
+  type SlashCommandOptionsOnlyBuilder
+} from "discord.js";
+import { genColor } from "../utils/colorGen";
+import { getLevel } from "../utils/database/levelling";
+import { getSetting } from "../utils/database/settings";
+import { errorEmbed } from "../utils/embeds/errorEmbed";
+import { imageColor } from "../utils/imageColor";
+
+export default class User {
+  data: SlashCommandOptionsOnlyBuilder;
+  constructor() {
+    this.data = new SlashCommandBuilder()
+      .setName("user")
+      .setDescription("Shows your (or another user's) info.")
+      .addUserOption(user => user.setName("user").setDescription("Select the user."));
+  }
+
+  async run(interaction: ChatInputCommandInteraction) {
+    const guild = interaction.guild!;
+    const target = guild.members.cache.get(
+      interaction.options.getUser("user")?.id ?? interaction.user.id
+    )!;
+
+    const user = await target.user.fetch();
+    let serverInfo = [`Joined on **<t:${Math.round(target.joinedAt?.valueOf()! / 1000)}:D>**`];
+    const guildRoles = guild.roles.cache.filter(role => target.roles.cache.has(role.id))!;
+    const avatar = target.displayAvatarURL();
+    const memberRoles = [...guildRoles].sort(
+      (role1, role2) => role2[1].position - role1[1].position
+    );
+    memberRoles.pop();
+    const rolesLength = memberRoles.length;
+
+    if (target.premiumSinceTimestamp)
+      serverInfo.push(`Boosting since **${target.premiumSinceTimestamp}**`);
+
+    if (memberRoles.length)
+      serverInfo.push(
+        `**${guildRoles.filter(role => target.roles.cache.has(role.id)).size! - 1}** ${
+          memberRoles.length == 1 ? "role" : "roles"
+        } • ${memberRoles
+          .slice(0, 5)
+          .map(role => `<@&${role[1].id}>`)
+          .join(", ")}${rolesLength > 3 ? ` and **${rolesLength - 3}** more` : ""}`
+      );
+
+    const embedColor =
+      user.hexAccentColor ?? (await imageColor(undefined, target)) ?? genColor(200);
+
+    let embed = new EmbedBuilder()
+      .setAuthor({
+        name: `${avatar ? "•  " : ""}${target.nickname ?? user.displayName}`,
+        iconURL: avatar
+      })
+      .setFields(
+        {
+          name: `<:discord:1266797021126459423> • Discord info`,
+          value: [
+            `Username is **${user.username}**`,
+            `Display name is ${
+              user.displayName == user.username ? "*not there*" : `**${user.displayName}**`
+            }`,
+            `Created on **<t:${Math.round(user.createdAt.valueOf() / 1000)}:D>**`
+          ].join("\n")
+        },
+        {
+          name: "📒 • Server info",
+          value: serverInfo.join("\n")
```

Comment:
What if they never joined the server ? (You can use a user id with the option attribute I think)

Reply:
thats a great question, will fix that